packages/thirdweb/src/react/core/hooks/contract/useReadContract.ts

```diff
@@ -18,8 +18,14 @@ import type {
 import type { PreparedMethod } from "../../../../utils/abi/prepare-method.js";
 import { getFunctionId } from "../../../../utils/function-id.js";
 import { stringify } from "../../../../utils/json.js";
+import type { Prettify } from "../../../../utils/type-utils.js";
 
-type PickedQueryOptions = Pick<UseQueryOptions, "enabled">;
+type PickedQueryOptions = Prettify<
+  Pick<UseQueryOptions, "enabled"> & {
+    refetchInterval?: number;
+    retry?: number;
+  }
```

Comment:
would it make more sense to just ad these to the `Pick` ?

Reply:
Initially tried that but react query had some janky types that were causing issues